src/file-events.c

```diff
@@ -196,17 +198,20 @@ static __always_inline void exit_mkdir(struct pt_regs *ctx, tail_call_slot_t tai
     fm->u.action.buffer_len = sizeof(file_message_t);
     fm->u.action.u.create.source_link = LINK_NONE;
 
-    cached_path->next_dentry = event.target_dentry;
+    init_filtered_cached_path(cached_path, event.target_dentry, event.target_vfsmount);
 
     ResolveName:
     ret = write_path(ctx, cached_path, &cursor, tail_call);
-    cached_path->next_dentry = NULL;
     if (ret < 0) goto EmitWarning;
+    if (cached_path->filter_state <= 0) goto NoEvent; // Didn't match watched path filter
 
-    write_null_char(cursor.buffer, cursor.offset);
+    bpf_probe_read_str(fm->u.action.filter_tag, MAX_TAG, cached_path->filter_tag);
     push_flexible_file_message(ctx, fm, *cursor.offset);
     // My attempts to call bpf_map_delete_elem() so far have caused the verifier to fail it
     // so just rely on the LRU functionality to keep the map from getting too big
+    write_null_char(cursor.buffer, cursor.offset);
```

Comment:
why do we write a null char into the buffer here? This buffer isn't going to be used anymore in this probe afaict and since it's per-cpu any subsequent probe should restart the cursor from the start.

Reply:
I was thinking we'd eventually use it to separate multiple paths in file messages with more than one.  I didn't think much about it and things apparently got re-arranged.  Will delete it and if it turns out we need something like that later we can add it back.